File: CC8_2/Main.py
from FindPath import *
import random

while True:
	x = (int)(input("Please input x: "))
	if x > 2:
		break

while True:
	y = (int)(input("Please input y: "))
	if y > 2:
		break

limitPoints = []
for i in range(max(x, y)):
	limitPoint = [random.randint(0, x - 1), random.randint(0, y -1)]
	if limitPoint in limitPoints:
		pass
	else:
		limitPoints.append(limitPoint)

print("limitPoints are")
for limitPoint in limitPoints:
	print(limitPoint)

findPath = FindPath(x, y, limitPoints)
print("Result is: ")
for path in findPath.startFind(0, 0):
	if path is "N":
		print("No Way")
		break
	print(path)

File: CC8_2/Main.java
import java.util.ArrayList;

public class Main {
	public static void main(String[] args) {
		int x = 5, y = 5;
		ArrayList<Point> limitPoints = new ArrayList<Point>();
		
		for (int i = 0; i < x; i += 3) {
			for (int j = 0; j < y; j += 2) {
				Point limitPoint = new Point((int)(Math.random() * (x - 1)), (int)(Math.random() * (y - 1)));
				if (limitPoints.contains(limitPoint) || (limitPoint.x == 0 && limitPoint.y == 0) || (limitPoint.x == x && limitPoint.y == y))
					continue;
				limitPoints.add(limitPoint);
			}
		}

		System.out.println("Limit points are :");
		for (Point limitPoint : limitPoints) {
			System.out.print("(" + limitPoint.x + ", " + limitPoint.y + ") ");
		}
		System.out.print("\n");

		FindPath fp = new FindPath(limitPoints, x, y);
		ArrayList<ArrayList<Point>> results = fp.getPaths(0, 0);
		System.out.println("Results of all path to the destination is :");
		if (results == null) 
			System.out.println("NO WAY");
		else {
			for (ArrayList<Point> result : results) {
				for (Point pathPoint : result) {
					System.out.print("(" + pathPoint.x + ", " + pathPoint.y + ") ");
				}
				System.out.print("\n");
			}
			System.out.print("\n");
		}
	}
}